src/arcsf/config/experiment.py

```diff
@@ -0,0 +1,304 @@
+import os
+import warnings
+from copy import copy
+from itertools import product
+
+import wandb
+import yaml
+from jinja2 import Environment, FileSystemLoader
+
+import arcsf.constants
+from arcsf.config.config import Config
+from arcsf.data.config import DataConfig
+from arcsf.models.config import ModelConfig
+
+
+def _listify(obj: object):
+    """Takes an object. If a list, returns it. If not, returns a list containing
+    the object.
+
+    Args:
+        obj: Object to ensure is a list.
+
+    Returns:
+        Either the original object if a list or a list containing the object.
+    """
+    if isinstance(obj, list):
+        return obj
+    return [obj]
+
+
+def _make_config_name(
+    top_config_name: str,
+    n: int,
+) -> str:
```

Comment:
Again minor but throughout this file short function names + args I'd prefer to be single line:
```suggestion
def _make_config_name(top_config_name: str, n: int) -> str:
```

If you remove the comma after the last arg `black` will auto-format them (or just re-add the comma if it wouldn't fit on a single line)

Reply:
Have done